ws-nextjs-app/pages/[service]/articles/handleArticleRoute.ts

```diff
@@ -0,0 +1,135 @@
+import { GetServerSidePropsContext } from 'next';
+import extractHeaders from '#server/utilities/extractHeaders';
+import { ARTICLE_PAGE, MEDIA_ARTICLE_PAGE } from '#app/routes/utils/pageTypes';
+import parseRoute from '#app/routes/utils/parseRoute';
+import nodeLogger from '#lib/logger.node';
+import { OK } from '#app/lib/statusCodes.const';
+import { ROUTING_INFORMATION } from '#app/lib/logger.const';
+import getPathExtension from '#app/utilities/getPathExtension';
+import PageDataParams from '#app/models/types/pageDataParams';
+import handleError from '#app/routes/utils/handleError';
+import { PageTypes, Toggles } from '#app/models/types/global';
+import augmentWithDisclaimer from '#app/routes/article/utils/augmentWithDisclaimer';
+import shouldRender from '#app/legacy/containers/PageHandlers/withData/shouldRender';
+import { ArticleMetadata } from '#app/models/types/optimo';
+import getPageData from '../../../utilities/pageRequests/getPageData';
+
+const logger = nodeLogger(__filename);
+
+const transformPageData = (toggles?: Toggles) =>
+  augmentWithDisclaimer({ toggles, positionFromTimestamp: 0 });
+
+const getDerivedArticleType = (metadata: ArticleMetadata) => {
+  let pageType: PageTypes = metadata?.type;
+
+  if (metadata?.type === 'article' && metadata?.consumableAsSFV) {
+    pageType = MEDIA_ARTICLE_PAGE;
+  }
+
+  return pageType;
+};
+
+export default async (context: GetServerSidePropsContext) => {
+  const {
+    resolvedUrl,
+    req: { headers: reqHeaders },
+  } = context;
+
+  const { service, renderer_env: rendererEnv } =
+    context.query as PageDataParams;
+
+  context.res.setHeader(
+    'Cache-Control',
+    'public, stale-if-error=90, stale-while-revalidate=30, max-age=30',
+  );
+
+  const urlWithoutQuery = resolvedUrl.split('?')?.[0];
```

Comment:
*nitpick*: Is it better to use URL here instead of splitting the string?

```suggestion
  const { href, search } = new URL(resolvedUrl);
  const urlWithoutQuery = href.replace(search, '');
```

Reply:
Yeah makes it neater and more readable, will change over

Reply:
Interesting this breaks, because the resolvedUrl is the pathname and the query params combined, which is weird:

```
 ⨯ TypeError: Invalid URL
    at default (pages/[service]/articles/handleArticleRoute.ts:47:28)
    at getServerSideProps (pages/[service]/articles/[[...variant]].page.tsx:33:21)
  45 |
  46 |   console.log(resolvedUrl); // eslint-disable-line no-console
> 47 |   const { href, search } = new URL(resolvedUrl);
     |                            ^
  48 |   const urlWithoutQuery = href.replace(search, '');
  49 |
  50 |   const { isAmp, isApp, isLite } = getPathExtension(urlWithoutQuery); {
  code: 'ERR_INVALID_URL',
  input: '/mundo/articles/ce42wzqr2mko.amp?renderer_env=test'
}
```

Slightly confusing as it's a next defined value, I would've thought it would be the full URL https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props#context-parameter